.github/workflows/test-all-warehouses.yml

```diff
@@ -44,3 +44,35 @@ jobs:
       dbt-version: ${{ inputs.dbt-version }}
       generate-data: ${{ inputs.generate-data || false }}
     secrets: inherit
+
+  notify_failures:
+    name: Notify Slack
+    needs: test
+    if: |
+      always() &&
+      contains(needs.test.result, 'failure') &&
```

Comment:
Why not use [`failure()`](https://docs.github.com/en/actions/learn-github-actions/expressions#failure)?

Reply:
It's step vs. jobs, not the same 
Not sure it'd work here